### PR TITLE
fix: Workflow of NPM Package release

### PR DESCRIPTION
**Pull request by surajgjadhav**
**Notes for Reviewers**

This PR fixes #
Updated workflow dispatch model for publishing NPM packages.

- Added option to choose release variant and release version 
- Base on selected release variant and version, it will update package version and create a build

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: .github/workflows/publish-npm-package.yml [hunk 1]
```diff
@@ -4,25 +4,43 @@
...
         required: true
+        default: 'prerelease'
+        type: choice
+        options:
+        - 'prerelease'
+        - 'patch'
+        - 'minor'
+        - 'major'
+      release_variant:
+        description: 'Release Version'
```
Comment (aabidsofi19) [inline .github/workflows/publish-npm-package.yml:20]: add a helpfull description

File: .github/workflows/publish-npm-package.yml [hunk 2]
```diff
@@ -4,25 +4,43 @@
...
-      contents: read
-      packages: write
+    permissions: write-all
     steps:
       - name: Checkout
         uses: actions/checkout@master
         with:
           ref: "master"
 
+      - name: Setup Git
```
Comment (aabidsofi19) [inline .github/workflows/publish-npm-package.yml:39]: we possibly dont need this , the commit action should take  care of this

**Comment by aabidsofi19** [comment]
@surajgjadhav  lets halt this effort for a while . there are other more  important items to take care of like sistent migration and playwright tests .

File: .github/workflows/publish-npm-package.yml [hunk 3]
```diff
@@ -4,25 +4,43 @@
...
+        run: |
+          git config user.name l5io
+          git config user.email ci@layer5.io
```
Comment (jameshorton2337) [inline .github/workflows/publish-npm-package.yml:42]:
```suggestion
```
Implementing @aabidsofi19's suggestion.

**Comment by jameshorton2337** [comment]
What is the name of this package to be?

File: .github/workflows/publish-npm-package.yml [hunk 4]
```diff
@@ -32,18 +50,36 @@ jobs:
...
+          echo "Updated Package Version" >> $GITHUB_STEP_SUMMARY
+          echo "info Old Version: ${OLD_VERSION}" >> $GITHUB_STEP_SUMMARY
+          echo "info New Version: ${NEW_VERSION}" >> $GITHUB_STEP_SUMMARY
+          
+      - name: Build Package
+        run: |
+          npm run build
+          echo "Build New Package" >> $GITHUB_STEP_SUMMARY
 
+      - name: Publich Package
```
Comment (jameshorton2337) [inline .github/workflows/publish-npm-package.yml:74]:
```suggestion
      - name: Publish Package
```

**Comment by leecalcote** [comment]
@surajgjadhav are you finishing this off?

File: .github/workflows/publish-npm-package.yml (hunk 1 above)
Comment (leecalcote) [reply .github/workflows/publish-npm-package.yml]: @surajgjadhav are you able to do this?

**Comment by leecalcote** [comment]
Should we consider this PR abandoned, @surajgjadhav?

**Comment by surajgjadhav** [comment]
Hi @leecalcote ,
@aabidsofi19 has told to hold on this PR so it was not merged yet
and there is one small change related to relese_varient option change (renaming 'no' option), that I will update and inform
Thanks

**Comment by surajgjadhav** [comment]
Hi @leecalcote , @aabidsofi19 ,
Updated PR 
Please Check
Thanks

File: .github/workflows/publish-npm-package.yml (hunk 1 above)
Comment (surajgjadhav) [reply .github/workflows/publish-npm-package.yml]: Done

**Comment by aabidsofi19** [comment]
@surajgjadhav  thanks for pushing the updates . if you would we have following confirmed. we can move ahead with this pr 
[x] Workflow gets triggered on repo release  
[x] Allows to override the version from action

File: .github/workflows/publish-npm-package.yml [hunk 5]
```diff
@@ -1,52 +1,79 @@
...
-          commit_options: "--signoff"
-          commit_message: "[Release]: Bump version to ${{ github.event.inputs.release-version || github.event.release.tag_name }}"
\ No newline at end of file
+# This workflow will run tests using node and then publish a package to GitHub Packages when a release is created
+# For more information see: https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages
+
+name: Publish NPM Package
+
+on:
+  workflow_dispatch:
```
Comment (leecalcote) [inline .github/workflows/publish-npm-package.yml:7]:
```suggestion
  release:
    types: [published]
  workflow_dispatch:
```
Adding trigger on release.

**Review by leecalcote — APPROVED** [review]

**Comment by leecalcote** [comment]
@aabidsofi19, I added an event trigger for release publish.

**Comment by leecalcote** [comment]
@aabidsofi19, the manual dispatch has the following inputs:

Screenshot 2024-08-22 at 9 18 25 PM

**Pull request merged**